### DISCUSSION TITLE
Release features for v2.0.0

### DISCUSSION THREAD

**Discussion by alexplesoiu** [alexplesoiu/dns-updater#10] [Ideas]
This will be the main thread for discussing the future releases for the project.
Here we can propose features, we can talk about to future of the project and so on.

First of all, for the next release, we should support all cloudflare domain settings (proxy, ttl, etc).
We should also organize the project and eliminate the misconfigurations. For example in .env we have DOMAINS_FILE_PATH=.\domains.json , but in our project we have [domains.example.json](https://github.com/alexplesoiu/dns-updater/blob/dev/domains.example.json) 

I also think we should put by default schedule every 5 minutes instead of 60 minutes. This way if something bad happen, like the router is going off and on and the ip is changing, we can update the DNS records in 5 minutes instead of 60 minutes.

We should also make the project easy to setup and run.

**Comment by SpasovskiFilip** [comment]
I think we can keep the `domains.example.json` file, just add more examples there as we have in the `README.md` file, and also add a `domains.json` file for use within the code.
And for the schedule, I agree, lest's make it at a 5 minute interval.

On a lower priority note, I was thinking of creating unit tests for the `main.py` code and running it on GitHub Actions. But the CF domain settings should be a bigger priority. I'll take a look at it tomorrow and see what i can add.

Regarding the easy setup and run, I liked the idea from PRs #8 & #9, we can expand on those. Unless you had something else in mind.

**Comment by alexplesoiu** [comment]
https://github.com/alexplesoiu/dns-updater/pull/9 has already been merged.
We can also expand this project to be helpful for frameworks like Django, Flask, FastAPI.
For `domains.example.json` file, we can just get rid of it, it can be quite confusing. Let's just add `domains.json`, with the full domain configuration. And we can add examples and tutorials into the `README.md` file. We can also setup a wiki page if needed. 

In the future releases we should provide a library that can be 'drag & drop' into any python project that uses networking. It will help a lot of projects I think.
The library should be easy to configure.

I will also setup the Github Actions for the project in weekend. 

As for docker, we can also add some support for env & arguments, so I can build the container and push it to Dockerhub. It will be interesting because then anybody can do just `docker pull` and then can just run the container with a specific config file / env vars.
We should also add more logs & checks. One can be with the github link for the tutorial if the configuration is bad or missing. 

The whole idea is to create a smooth experience for anybody who wants to use this :)

**Reply by SpasovskiFilip** [reply]
Hi @alexplesoiu I added some of the changes we talked about in PR #11
